doc/compiler/optimizer/IntroReadLogFile.md

```diff
@@ -0,0 +1,1188 @@
+ n4932n   (  2)      ==>iRegLoad (in GPR_0545) (SeenRealReference )
+ n4933n   (  2)      ==>aRegLoad (in &GPR_0560) (X!=0 X>=0 SeenRealReference sharedMemory )
+ n4934n   (  1)      ==>aRegLoad (in &GPR_0561)
+------------------------------
+------------------------------
+ n1862n   (  0)  ifacmpeq --> block_193 BBStart at n1836n ()                                          [0x7fef25c7d5c0] bci=[9,1,-] rc=0 vc=4181 vn=- li=479 udi=- nc=3 flg=0x20
+ n1858n   (  2)    ==>l2a (in &GPR_0562) (X!=0 sharedMemory )
+ n1861n   (  0)    aconst NULL (X==0 sharedMemory )                                                   [0x7fef25c7d570] bci=[9,1,-] rc=0 vc=4181 vn=- li=479 udi=- nc=0 flg=0x2
+ n4935n   (  1)    GlRegDeps ()                                                                       [0x7fef26429640] bci=[9,1,-] rc=1 vc=4181 vn=- li=479 udi=- nc=6 flg=0x20
+ n4929n   (  1)      ==>iRegLoad (in GPR_0512) (SeenRealReference )
+ n4930n   (  3)      ==>iRegLoad (in GPR_0528) (cannotOverflow SeenRealReference )
+ n4931n   (  1)      ==>aRegLoad (in &GPR_0544) (SeenRealReference sharedMemory )
+ n4932n   (  1)      ==>iRegLoad (in GPR_0545) (SeenRealReference )
+ n4933n   (  1)      ==>aRegLoad (in &GPR_0560) (X!=0 X>=0 SeenRealReference sharedMemory )
+ n4934n   (  0)      ==>aRegLoad (in &GPR_0561)
+------------------------------
+
+ [0x7fef12b0abc0]	test	&GPR_0562, &GPR_0562		# TEST8RegReg
+ [0x7fef12b0b080]	assocreg                        # assocreg
+	POST: [&GPR_0544 : ebx] [GPR_0512 : edi] [GPR_0528 : r8d] [GPR_0545 : r9d] [&GPR_0560 : r12d] [&GPR_0561 : r13d]
+ [0x7fef12b0ae10]	je	Label L0025			# JE4
+	 PRE: [&GPR_0561 : r13d] [&GPR_0560 : r12d] [GPR_0545 : r9d] [&GPR_0544 : ebx] [GPR_0528 : r8d] [GPR_0512 : edi]
+	POST: [&GPR_0561 : r13d] [&GPR_0560 : r12d] [GPR_0545 : r9d] [&GPR_0544 : ebx] [GPR_0528 : r8d] [GPR_0512 : edi]
+
+…
+============================================================
+; Live regs: GPR=5 FPR=0 VRF=0 {&GPR_0562, GPR_0545, &GPR_0544, GPR_0528, GPR_0512}
+------------------------------
+ n4736n   (  0)  astore  <temp slot 8>[#522  Auto] [flags 0x4007 0x0 ]                                [0x7fef26425810] bci=[9,1,-] rc=0 vc=4181 vn=- li=356 udi=- nc=1
+ n1858n   (  2)    ==>l2a (in &GPR_0562) (X!=0 sharedMemory )
+------------------------------
+------------------------------
+ n4736n   (  0)  astore  <temp slot 8>[#522  Auto] [flags 0x4007 0x0 ]                                [0x7fef26425810] bci=[9,1,-] rc=0 vc=4181 vn=- li=356 udi=- nc=1
+ n1858n   (  1)    ==>l2a (in &GPR_0562) (X!=0 sharedMemory )
+------------------------------
+
+ [0x7fef12b0bdd0]	mov	qword ptr [vfp], &GPR_0562		# S8MemReg, SymRef  <temp slot 8>[#678  Auto] [flags 0x4007 0x0 ]
+…
+```
+
+`n1858n l2a` is evaluated and the value is stored in `&GPR_0562` in the instruction at `0x7fef12b0a7c0`.
+`&` means the virtual register is a collected reference. The reference needs to be conveyed in a garbage collector.
```

Comment:
Not sure if people will recognize what collected reference means... may want to elaborate on that and its relationship to GC.

Reply:
Good point! @joransiu I'm not sure what the best way is to describe collected references. Would this make sense:  collected references are heap objects that are referenced by more than one BB and that need to be conveyed in GC?

Reply:
I think a "collected reference" is a more fundamental notion that is tied to any reference value as such, i.e. independent of which BB it gets referenced from. If a reference typed node in our IL could result in a value that is on the Java heap, then it must be marked as "collected" implying that the GC would need to be notified about it if the value is live across a GC point. Note that such a collected reference may be held live within a single BB and yet be live across a GC point (since a single block can contain any number of GC points). "Collected references" in the IL get associated with a virtual register or stack slot that is in turn considered collected when codegen happens and this information eventually gets used to emit GC stack and register maps at the GC points notifying the GC about live objects as well as reference slots/registers needing to be updated if the object is moved.

Reply:
Addressed in 9c5fd3b